shared_web/static/js/pd.js

```diff
@@ -367,6 +368,70 @@ PD.htmlEscape = function (s) {
     return $("<div>").text(s).html();
 };
 
+PD.Filter = {};
+
+PD.Filter.init = function () {
+    // Apply the filter with the initial value of the form
+    PD.Filter.scryfallFilter($("#scryfall-filter-input").val())
+
+    // set up the event handlers for the form
+    $("#scryfall-filter-form").submit(function () {
+        PD.Filter.scryfallFilter($("#scryfall-filter-input").val());
+        return false;
+    });
+    $("#scryfall-filter-reset").click(PD.Filter.reset);
+};
+
+PD.Filter.scryfallFilter = function (query) {
+    if (query == ""){
+        PD.Filter.reset();
+        return;
+    }
+
+    $("#scryfall-filter-submit").attr("disabled", "disabled").text("Loading...");
+    $("#scryfall-filter-reset").attr("disabled", "disabled").text("Loading...");
+    $("#scryfall-filter-form").submit(function () { return false; })
+
+    query = "f:pd and (" + query + ")";
+    card_names = [];
+
+    function doFilter () {
+        $("tr.cardrow").each( function () {
+            jqEle = $(this)
+            if (card_names.indexOf(this.id.split('-')[1]) == -1){
+                jqEle.hide();
+            }
+            else {
+                jqEle.show();
+            }
+        });
+        $("#scryfall-filter-submit").removeAttr("disabled").text("Search");
+        $("#scryfall-filter-reset").removeAttr("disabled").text("Reset");
+        $("#scryfall-filter-form").submit(function () { return false; })
+    }
+
+    function parse_and_continue (blob) {
+        // TODO: error handling
```

Comment:
Prefer GitHub issues to TODOs. This and a couple of other things on this review should be found by the linter, I think. Try `make lint` in the root of the project on your dev machine. Maybe that skips js?

Reply:
Sorry, yeah, the linter didn't catch that.